cypress/utils/apps/query_enhancements/export_as_csv.js

```diff
@@ -0,0 +1,161 @@
+ * @property {boolean} hasTime - whether the dataset has time
+ */
+
+/**
+ * Returns the ExportAsCsvTestConfig
+ * @returns {ExportAsCsvTestConfig[]}
+ */
+export const generateExportAsCsvIndexPatternTestConfigurations = () => {
+  return Object.values(DatasetTypes)
+    .map((dataset) => {
+      const sharedConfig = {
+        datasetType: dataset.name,
+        languages: dataset.supportedLanguages,
+      };
+      return [
+        {
+          ...sharedConfig,
+          dataset: `${INDEX_WITH_TIME_1}${dataset.name === DatasetTypes.INDEXES.name ? '' : '*'}`,
+          hasTime: true,
+          saveName: `${dataset.name}-Time`,
+        },
+        {
+          ...sharedConfig,
+          dataset: `${INDEX_WITHOUT_TIME_1}${
+            dataset.name === DatasetTypes.INDEXES.name ? '' : '*'
+          }`,
+          hasTime: false,
+          saveName: `${dataset.name}-NoTime`,
+        },
+      ];
+    })
+    .flat();
+};
+
+/**
+ * Returns the number of queried results we should have for a given language
+ * @param {QueryEnhancementLanguageData} language - the language data
+ * @returns {number}
+ */
+export const getQueriedCountForLanguage = (language) => {
+  switch (language.name) {
+    case QueryLanguages.DQL.name:
+      return 500;
+    case QueryLanguages.Lucene.name:
+      return 500;
+    case QueryLanguages.SQL.name:
+      return 10;
+    case QueryLanguages.PPL.name:
+      return 1000;
+  }
+};
+
+/**
+ * Returns what the header should be for the downloaded file with source
+ * @property {boolean} hasTime - whether the dataset has time
+ * @returns {string[]}
+ */
+export const getHeadersForSourceDownload = (hasTime) => {
+  if (hasTime) {
+    return ['timestamp', '_source'];
+  } else {
+    return ['_source'];
+  }
+};
+
+/**
+ * Returns what the header should be for the downloaded file with selected fields
+ * @property {boolean} hasTime - whether the dataset has time
+ * @returns {string[]}
+ */
+export const getHeadersForDownloadWithFields = (hasTime) => {
+  if (hasTime) {
+    return ['timestamp', 'bytes_transferred', 'personal.name'];
+  } else {
+    return ['bytes_transferred', 'personal.name'];
+  }
+};
+
+/**
+ * Returns what the first row should be for the downloaded file for file with _source
+ * @param {QueryEnhancementLanguageData} language - the language data
+ * @returns {string}
+ */
+export const getFirstRowTimeForSourceDownload = (language) => {
+  let time = 'Dec 31, 2022 @ 14:14:42.801';
+
+  if ([QueryLanguages.SQL.name, QueryLanguages.PPL.name].includes(language.name)) {
+    time = 'Jan 1, 2021 @ 00:00:00.000';
+  }
+
+  return time;
+};
+
+/**
+ * Returns what the first row should be for the downloaded file for file with selected fields
+ * @param {QueryEnhancementLanguageData} language - the language data
+ * @param {boolean} hasTime - whether the dataset has time
+ * @returns {string[]}
+ */
+export const getFirstRowForDownloadWithFields = (language, hasTime) => {
```

Comment:
Same as above. I may be misunderstanding though if this is for testing purposes only.

Reply:
Yeah this is purely for testing purposes. I am ensuring that the formatting of the fields and the columns are correct in CSV